HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,711 @@
+status: Draft
+created: 2024-2-25
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/905
+updated: 
+requires: 
+replaces: 655, 777
+superseded-by: 
+---
+
+## Abstract
+
+This HIP enables frictionless airdrops of both fungible and non-fungible tokens by removing the requirement to pre-associate tokens with the receiver's account, and by allowing the sender to create associations on the receiver's account automatically at transfer time by paying for any token-associations created as a result of the airdrop. The sovereignty of account holders who do not wish to receive unsolicited tokens is preserved by supporting the concept of pending token transfers to such an account. Potential token receivers may claim their token airdrop while available if desired and airdrop senders may cancel their sent airdrop should they change their mind.
+
+With this HIP, accounts will no longer prepay for unused token association slots. Association fees will be charged for those slots only when a token is actually associated with an account.
+
+This HIP also introduces the concept of token rejection. Here an undesired token class value is returned to the treasury with all custom fees waived.
+
+## Motivation
+
+The concept of `token associations` was added to Hedera to protect the network and its users. [HIP-23] (https://hips.hedera.com/hip/hip-23) lists some of the reasons for introducing token associations.
+
+However, mandatory token pre-association — where a recipient must associate the token-type before receiving a token — contributes significant friction for many web3 use cases, especially those where the user has a reasonable desire to receive a token without taking any action. There has been a strong ask from the community to reduce this friction in a cost effective and intuitive manner.
+
+Users who wish to *airdrop* (send unsolicited) tokens encounter friction in several ways:
+
+- They must convince users to pre-associate with their token, which is a poor user experience for users who wish to hold the token.
+- If they attempt to airdrop tokens widely, many transactions will simply fail because many accounts have not pre-associated with the token.
+
+Users who wish to receive airdrops:
+
+- Must prepay for auto association slots, which are expensive, even when not filled.
+- Users must hold hbars in order to accept an airdrop (to pay for the association), making it difficult in many workflows to create and pre-populate accounts with tokens.
+
+Retail users from other EVM chains are surprised because other chains do not have the concept of token association. As such, users are accustomed to receiving unsolicited tokens and viewing them in their wallet. Non-Hedera wallets have no native method for understanding token association. Similarly, senders are not used to coordinating token association in advance. This currently provides friction and may serve as a blocker to adoption by developers coming from other chains.
```

Comment:
Equally there are many voices in the web3 community and from other networks that want to be able to be in control of what lands in their wallet.

Reply:
Can add this sentence

Reply:
Let me know if you meant in replacement

Reply:
In addition, not replacement, the original statement is entirely valid, just adding colour.